### DISCUSSION TITLE
Is it possible to get comment count?

### DISCUSSION THREAD

**Discussion by iacchus** [giscus/giscus#113] [Q&A] [answered]
Hello,

thanks for the awesome software! :)

Is it possible to get the comments counting for blog post?

**Comment by laymonage** [comment]
Could you elaborate on your use case? The comments count is already shown by giscus at the top of the widget.

**Reply by iacchus** [reply]
The home page at the blog. Like disqus, a link that shows `0 Comments` or `1 Comment` or `[n] Comments`

I am implementing it in the following blog of mine, I would like also to set the button to open the sidebar with "[number of] Comments." (sorry for being still in development.) Here in this blog post for example:

https://iacchus.github.io/md-test.html

**Comment by laymonage** [comment, marked as answer]
Hey @iacchus, this is now possible by retrieving `IMetadataMessage` in a `message` event listener and accessing [`giscusData.discussion.totalCommentCount`](https://github.com/laymonage/giscus/blob/main/lib/types/giscus.ts#L24). You can follow [this guide](https://github.com/laymonage/giscus/blob/main/ADVANCED-USAGE.md#giscus-to-parent-message-events) for more details.

You need to enable the "Emit discussion metadata" feature in the configuration:
![image](https://user-images.githubusercontent.com/6379424/125184772-b5c55d80-e24a-11eb-8922-155e19e69a3e.png)

or add `data-emit-metadata="1"` to your existing `<script>` tag.

**Reply by iacchus** [reply]
Very nice. Is there a way of getting the comments of all the posts of the page? (for example, the blog's home page.)

**Reply by laymonage** [reply]
Unfortunately no, unless you embed giscus for every post that appears on the page.

Another hacky way to do it would be to embed giscus once, then [change the discussion config](https://github.com/laymonage/giscus/blob/main/ADVANCED-USAGE.md#parent-to-giscus-message-events) inside a loop, while listening to the metadata. But this is very hacky and not sustainable. I think your best bet would be to call GitHub's API on your own.
